### PR TITLE
Updating documentation on shared variables

### PR DESCRIPTION
**Pull request by 317070**
You can make a pull request to Theano github repo in order to fix a bug or add new features.
Once a pull request is submitted, [Travis](https://travis-ci.org/) is launched to verify the pull request
given the current stack of unit-tests in Theano.

Updating documentation on shared variables.

### CODE REVIEW

File: doc/library/compile/shared.txt [hunk 1]
```diff
@@ -19,11 +19,27 @@
...
+    
+    .. method:: set_value(self, new_value, borrow=False)
+       :param new_value: 
+       :type new_value:
+       
+       :param borrow: True to use the new_value directly, potentially creating problems
+            related to aliased memory.
+       :type borrow: bool
+       
+        Changes to this value will be visible to all functions using this SharedVariable.
```
Comment (nouiz) [inline doc/library/compile/shared.txt]: This is making warning during documentation generation: 

/home/travis/build/Theano/Theano/doc/library/compile/shared.txt:33: ERROR: Error in "method" directive:
```
invalid option data: extension option field name may not contain multiple words.

.. method:: set_value(self, new_value, borrow=False)

   :param new_value:

   :type new_value:

   :param borrow: True to use the new_value directly, potentially creating problems

        related to aliased memory.

   :type borrow: bool

    Changes to this value will be visible to all functions using this SharedVariable.
```

This need to be fixed before merging.

thanks.

File: doc/library/compile/shared.txt [hunk 2]
```diff
@@ -19,11 +19,27 @@
...
+    .. method:: get_value(self, borrow=False, return_internal_type=False)
+    
+       :param borrow: True to permit returning of an object aliased to internal memory.
+       :type borrow: bool
+               
+       :param return_internal_type: True to permit the returning of an arbitrary type object used
+               internally to store the shared variable.
+       :type return_internal_type: bool
+       
+       Changes to this value will be visible to all functions using this SharedVariable.
```
Comment (abergeron) [inline doc/library/compile/shared.txt]: This isn't true unless you set borrow=True (and may not be true depending on whether you also set return_internal_type=True).

File: doc/library/compile/shared.txt [hunk 3]
```diff
@@ -19,11 +19,27 @@
...
+    
+    .. method:: set_value(self, new_value, borrow=False)
+       :param new_value: The new value.
+       :type new_value: A compatible type for this shared variable.
+       
+       :param borrow: True to use the new_value directly, potentially creating problems
+            related to aliased memory.
+       :type borrow: bool
+       
+       Changes to this value will be visible to all functions using this SharedVariable.
```
Comment (abergeron) [inline doc/library/compile/shared.txt]: Changes to what value exactly?

**Comment by nouiz** [comment]
fixed

**Pull request closed without merging**